### PR TITLE
Add hover_tooltips, hover_mode, hover_formatters opts to easily modify hover

### LINKED ISSUE #1816: Easy specification of dimensions for hover tool
Instead of building an explicit hover tool to choose dimensions to show hover information for, maybe we could have a syntax such as:

```
tools=['hover+time|voltage']
```

Where ``+`` means you are building a whitelist (what to include) and ``-`` means it is a blacklist (what to exclude). E.g to exclude 'time' and 'voltage':

```
tools=['hover-time|voltage']
```

### PR DESCRIPTION
**Pull request by ahuang11**
See discussion at https://github.com/holoviz/holoviews/issues/1816#issuecomment-2043926943

https://github.com/holoviz/holoviews/assets/15331990/0702757e-d4f1-421f-8975-689598bd219f

resolves #1816

### CODE REVIEW

**Comment by ahuang11** [comment]
Yeah I forgot to do so this time and I've already lost the code for that, but the edited notebook should have everything you need to try out all the functionality

**Comment by droumis** [comment]
Looks great with `subcoordinate_y`! Nice work. Not a big deal but the only thing I couldn't figure out yet is how to have the unit included in the tooltip when the unit is set in the hv.Dimension and a formatter is applied in hover_tooltips, like {0.2f} (see 'amplitude' vs 'time'). It's not a big deal to add the unit it manually though.

<details> <summary> Code </summary>

```python
import numpy as np
import holoviews as hv; hv.extension('bokeh')

n_channels = 4
n_seconds = 300
fs = 256
init_freq = .01  # Initial sine wave frequency in Hz
freq_inc = 2/n_channels  # Frequency increment
total_samples = n_seconds * fs
time = np.linspace(0, n_seconds, total_samples)
channels = [f'CH {i}' for i in range(n_channels)]

data = np.array([np.sin(2 * np.pi * (init_freq + i * freq_inc) * time)
                 for i in range(n_channels)])
# print(f'shape: {data.shape} (n_channels, samples) ')

time_dim = hv.Dimension('Time', unit='s')
amplitude_dim = hv.Dimension('Amplitude', unit='µV')

curves = []
for channel, channel_data in zip(channels, data):
    ds = Dataset((time, channel_data), [time_dim, amplitude_dim])
    curve = hv.Curve(ds, time_dim, amplitude_dim, group='EEG', label=f'{channel}')
    curve.opts(
        subcoordinate_y=True,
        subcoordinate_scale=.75,
        color="black",
        line_width=1,
        tools=['hover'],
        hover_tooltips=[
            ("Group", "$group"),
            ("Channel", "$label"),
            "Time",
            ("Amplitude", "@Amplitude{0.2f}")],
        # hover_mode = 'vline',
        )
    curves.append(curve)

curves_overlay = hv.Overlay(curves, kdims="Channel")
curves_overlay = curves_overlay.opts(
    ylabel="Channel",
    show_legend=False,
    aspect=3.5,
    responsive=True,
    shared_axes=False,
    xlim=(50,150),
    title='Multi-Channel Timeseries',
)

curves_overlay

```

</details>

image


and some fun with `hover_mode = 'vline'`:

https://github.com/holoviz/holoviews/assets/6613202/057d5af4-b339-4607-a6db-ea879e05a6a7

**Comment by ahuang11** [comment]
Ah, right as you commented, I think I fixed it in the newest push. Let me also try...

**Comment by ahuang11** [comment]
Okay it should work with your code now!

image

**Comment by ahuang11** [comment]
> and some fun with hover_mode = 'vline':

Would be a nice Bokeh feature to have to have like hover_align="left", "center", "right"

File: holoviews/plotting/bokeh/element.py [hunk 1]
```diff
@@ -286,16 +295,139 @@ def _hover_opts(self, element):
         dims += element.dimensions()
         return list(util.unique_iterator(dims)), {}
 
+    def _replace_label_group(self, element, tooltip):
```
Comment (Hoxbro) [inline holoviews/plotting/bokeh/element.py:298]: All these methods, in my opinion, need to start with `_hover` to understand that they are easily related to the hover tool.

**Comment by ahuang11** [comment]
Should be ready for re-review if needed.

The UI tests passes in my local machine, but I noticed it failed here. I added flaky markers.

**Comment by Hoxbro** [comment]
For the flaky tests, try using `wait_until`.

File: holoviews/tests/ui/bokeh/test_hover.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    page = serve_hv(hv_image)
+    hv_plot = page.locator(".bk-events")
+    wait_until(lambda: expect(hv_plot).to_have_count(1))
+    bbox = hv_plot.bounding_box()
+
+    # Hover over the plot
+    page.mouse.move(bbox["x"] + bbox["width"] / 2, bbox["y"] + bbox["height"] / 2)
+    page.mouse.up()
+
+    wait_until(lambda: expect(page.locator(".bk-Tooltip")).to_have_count(1))
```
Comment (Hoxbro) [inline holoviews/tests/ui/bokeh/test_hover.py:33]: Try adding page to wait until.

**Comment by ahuang11** [comment]
Eh the test just refuses to pass...

File: examples/user_guide/Plotting_with_Bokeh.ipynb [hunk 3]
```diff
@@ -695,7 +714,7 @@
     "\n",
     "points = hv.Points(\n",
     "    elements, ['electronegativity', 'density'],\n",
-    "    ['name', 'symbol', 'metal', 'CPK', 'atomic radius']\n",
+    "    ['name', 'symbol', 'metal', 'CPK', 'atomic radius'],\n",
```
Comment (Hoxbro) [inline examples/user_guide/Plotting_with_Bokeh.ipynb:717]: I think we should move the section from Bokeh.models import HoverTool` to the end of this section. It is the last resort when things do not work.

However, we should at least mention early that it uses the model behind the scenes and links to Bokeh's documentation.

File: examples/user_guide/Plotting_with_Bokeh.ipynb [hunk 4]
```diff
@@ -14,7 +14,6 @@
    "outputs": [],
    "source": [
     "import numpy as np\n",
-    "import pandas as pd\n",
```
Comment (Hoxbro) [inline examples/user_guide/Plotting_with_Bokeh.ipynb:17]: I think this should be added back

**Review by Hoxbro — APPROVED** [review]

**Pull request merged**